### PR TITLE
feat(sindi): add parameter check for sindi

### PR DESCRIPTION
**Pull request by inabao**
## Summary by Sourcery

Enforce validation on SINDI index build and search parameters, handle invalid inputs gracefully, and expand tests accordingly

New Features:
- Add runtime checks for doc_prune_ratio and window_size bounds in SINDIParameter
- Add runtime checks for term_prune_ratio, query_prune_ratio, and n_candidate bounds in SINDISearchParameter and KnnSearch
- Return failed IDs for zero‐length vectors in SINDI::Add instead of silently ignoring them

Bug Fixes:
- Fix inner search ef selection logic to ensure ef is at least k

Enhancements:
- Raise default window_size from 100 to 10000
- Modify TestFactory to return nullptr on expected failures

Tests:
- Add tests for invalid build and search parameters and zero‐length vector insertion scenario

### CODE REVIEW

**Review by ShawnShawnYou — APPROVED** [review]
LGTM

File: src/algorithm/sindi/sindi.cpp [hunk 1]
```diff
@@ -65,6 +66,11 @@ SINDI::Add(const DatasetPtr& base) {
         auto cur_window = cur_element_count_ / window_size_;
         auto window_start_id = cur_window * window_size_;
         const auto& sparse_vector = sparse_vectors[i];
+        if (sparse_vectors->len_ <= 0) {
+            failed_ids.push_back(ids[i]);
+            logger::warn("query.len_ ({}) is invalid for id ({})", sparse_vectors->len_, ids[i]);
```
Comment (wxyucs) [inline src/algorithm/sindi/sindi.cpp]: `query.len_` -> `sparse_vectors->len_`

File: src/algorithm/sindi/sindi.cpp [hunk 2]
```diff
@@ -98,15 +104,19 @@ SINDI::KnnSearch(const DatasetPtr& query,
     const auto* sparse_vectors = query->GetSparseVectors();
     CHECK_ARGUMENT(query->GetNumElements() == 1, "num of query should be 1");
     auto sparse_query = sparse_vectors[0];
+    CHECK_ARGUMENT(sparse_vectors->len_ > 0,
+                   fmt::format("query.len_ ({}) is invalid", sparse_vectors->len_));
```
Comment (wxyucs) [inline src/algorithm/sindi/sindi.cpp]: ditto

File: src/algorithm/sindi/sindi_parameter.cpp [hunk 3]
```diff
@@ -23,6 +23,8 @@ void
 SINDIParameter::FromJson(const JsonType& json) {
     if (json.contains(SPARSE_DOC_PRUNE_RATIO)) {
         doc_prune_ratio = json[SPARSE_DOC_PRUNE_RATIO];
+        CHECK_ARGUMENT((doc_prune_ratio >= 0.0F and doc_prune_ratio <= 0.5F),
```
Comment (wxyucs) [inline src/algorithm/sindi/sindi_parameter.cpp]: `0.0F <= doc_prune_ratio and doc_prune_ratio <= 0.5F`

Comment (inabao) [reply src/algorithm/sindi/sindi_parameter.cpp]: done

File: src/algorithm/sindi/sindi.cpp (hunk 2 above)
Comment (inabao) [reply src/algorithm/sindi/sindi.cpp]: done

File: src/algorithm/sindi/sindi.cpp (hunk 1 above)
Comment (inabao) [reply src/algorithm/sindi/sindi.cpp]: done

**Review by wxyucs — APPROVED** [review]
lgtm
